Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb

```diff
@@ -159,7 +168,15 @@ def process(queue)
             start_index = 0 if start_index < 0
           end
 
-          blob, content = @azure_blob.get_blob(@container, blob_name, {:start_range => start_index} )
+          blob_size = blob.properties[:content_length]
+          end_index = start_index + MAX_BUFFER_SIZE - 1
+          @sleep_interval = 0
```

Comment:
An cleaner approach may be, have process() return a bool which represent "if I am still in the middle of a blob", and the caller checks and decides to sleep or not based on that. That way, you don't need this extra member called @sleep_internal and setting it to 0 or @interval.